### PR TITLE
Support for deleting container

### PR DESCRIPTION
**Pull request by YaoZengzeng**

### CODE REVIEW

File: serverrpc/container.go [hunk 1]
```diff
@@ -33,3 +33,16 @@ func (s *ServerRPC) ContainerStop(c context.Context, req *types.ContainerStopReq
 
 	return &types.ContainerStopResponse{}, nil
 }
+
+// ContainerRemove implements DELETE /container
```
Comment (feiskyer) [inline serverrpc/container.go]: Update doc as `ContainerRemove deletes the specified container`

File: client/api/interface.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ type APIInterface interface {
 	ContainerLogs(container, since string, timestamp, follow bool, tail string) (io.ReadCloser, string, error)
 	KillContainer(container string, sig int) error
 	StopContainer(container string) error
+	RmContainer(container string) error
```
Comment (feiskyer) [inline client/api/interface.go]: Rename `RmContainer` to full name? e.g. `RemoveContainer `

File: daemon/rm.go [hunk 3]
```diff
@@ -74,3 +77,78 @@ func (daemon *Daemon) RemovePodContainer(p *Pod) {
...
+		return err
+	}
+
+	r, err := daemon.ContainerInspect(containerId, false, version.Version("1.21"))
+	if err != nil {
+		return err
+	}
+
+	if err := daemon.Daemon.ContainerRm(containerId, &dockertypes.ContainerRmConfig{}); err != nil {
+		return fmt.Errorf("failed to rm container %s", containerId)
```
Comment (gao-feng) [inline daemon/rm.go]: return or print out err returned by ContainerRm

File: daemon/rm.go [hunk 4]
```diff
@@ -74,3 +77,78 @@ func (daemon *Daemon) RemovePodContainer(p *Pod) {
...
+	}
+
+	rsp, ok := r.(*dockertypes.ContainerJSON)
+	if !ok {
+		err = fmt.Errorf("fail to unpack container json response for %s of %s", containerId, pod.Id)
+		return err
```
Comment (gao-feng) [inline daemon/rm.go]: return fmt.Errorf(...) directly.

File: daemon/rm.go [hunk 5]
```diff
@@ -74,3 +77,78 @@ func (daemon *Daemon) RemovePodContainer(p *Pod) {
...
+	jsons, err := pod.TryLoadContainers(daemon)
+	if err != nil {
+		return err
+	}
+	if err = pod.ParseContainerJsons(daemon, jsons); err != nil {
+		glog.Errorf("Found an error while parsing the Containers json: %v", err)
+		return err
+	}
+
+	daemon.PodList.Put(pod)
```
Comment (gao-feng) [inline daemon/rm.go]: no need to put pod twice.

Comment (YaoZengzeng) [inline daemon/rm.go]: It's necessary.The first time before "daemon.WritePodAndContainers(pod.Id)", we need put pod to update pod status.Then the spec of pod also being changed,so we need to update again to keep pod fresh.

**Comment by YaoZengzeng** [comment]
Ready for review, after the pr of "runv #267"  and "hyperstart #121" being merged.

File: daemon/rm.go (hunk 5 above)
Comment (gao-feng) [inline daemon/rm.go]: podlist.Put updates the containers array in podlist while daemon.WritePodAndContainers uses podstatus.Containers to updates the database, the podstatus.Containers was already updated.

File: daemon/rm.go [hunk 6]
```diff
@@ -74,3 +77,79 @@ func (daemon *Daemon) RemovePodContainer(p *Pod) {
...
 	daemon.db.DeleteP2C(p.Id)
 }
+
+func (daemon *Daemon) RemoveContainer(containerId string) error {
+	pod, idx, ok := daemon.PodList.GetByContainerIdOrName(containerId)
+	if !ok {
+		return fmt.Errorf("can not find container %s", containerId)
+	}
+
+	if !pod.TransitionLock("rm") {
```
Comment (gao-feng) [inline daemon/rm.go]: pod.TransitionLock("rmContainer")

**Comment by gnawux** [comment]
I think we may close this PR, remove container from pod is supported in #445 and #469

**Pull request closed without merging**